justfile

```diff
@@ -0,0 +1,133 @@
+# See https://just.systems/man/en
+
+# By default we compile in development mode mode because it's faster.
+rs-build-type := if env_var_or_default("RELEASE", "") == "" { "debug" } else { "release" }
+
+# Overriddes the default Rust toolchain version.
+rs-toolchain := ""
+
+rs-features := 'all'
+
+_cargo := "cargo" + if rs-toolchain != "" { " +" + rs-toolchain } else { "" }
+
+archs := 'linux/amd64,linux/arm64,linux/arm/v7'
+
+# Check that the Rust code is formatted correctly.
+rs-check-fmt:
+    {{ _cargo }} fmt --all -- --check
+
+# Fetch Rust dependencies.
+rs-fetch:
+    {{ _cargo }} fetch --locked
+
+# Lint Rust code.
+rs-clippy:
+    {{ _cargo }} clippy --frozen --workspace --all-targets --no-deps {{ _features }} {{ _fmt }}
+
+# Generate Rust documentation.
+rs-doc *flags:
+    {{ _cargo }} doc --frozen \
+        {{ if rs-build-type == "release" { "--release" } else { "" } }} \
+        {{ _features }} \
+        {{ flags }}
+
+rs-test-build:
+    {{ _cargo-test }} --no-run --frozen \
+        {{ _features }} \
+        {{ _fmt }}
+
+# Run Rust unit tests
+rs-test *flags:
+    {{ _cargo-test }} --frozen \
+        {{ if rs-build-type == "release" { "--release" } else { "" } }} \
+        {{ _features }} \
+        {{ flags }}
```

Comment:
Not a blocker but perhaps we could just use these recipes as wrappers around `just-cargo`? It seems that some of them already exist in there, and the assumption is that we would use it in a dev container (or have `just-cargo` on your path if you vim). https://github.com/linkerd/dev/blob/main/bin/just-cargo#L74-L101

Could also be done as a follow-up, not sure if it's worth spending a bunch of time on the justfile here.

Reply:
Thanks @mateiidavid . I did give `just-cargo` a try like you suggested, but couldn't figure how to solve the issue of `clang` not being found. I'm gonna move forward for the time being, and we can revisit this as a follow-up :pray: